File: Assets/My/Scripts/LCallC/LuaCallStatic.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

namespace YY
{
    public static class TestStatic
    {
        public static int ID = 78;
        public static string Name { get; set; }

        public static string Output()
        {
            return "Static";
        }
        public static void Default(string str = "abc")
        {
            Debug.Log(str);
        }
    }
}


public class LuaCallStatic : MonoBehaviour
{
    void Start()
    {
        xLuaEnv.Instance.Dostring("require('LCallC/LuaCallStatic')");
    }

    private void OnDestroy()
    {
        xLuaEnv.Instance.Free();
    }
}

File: Assets/StreamingAssets/Lua/LCallC/LuaCallStatic.lua
print("Lua 调用静态类")
--规则"CS.命名空间.类名.成员变量"

--静态成员变量
print(CS.YY.TestStatic.ID);
CS.YY.TestStatic.Name = "admin";

--静态成员方法
print(CS.YY.TestStatic.Output());
CS.YY.TestStatic.Default();
CS.YY.TestStatic.Default("Hello I Call U");

